analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py

```diff
@@ -448,107 +412,92 @@ def get_query(source: ObjectStream) -> ObjectStream:
-# - when set to `coffea`, processing will happen with pure `coffea`,
-# - if `PIPELINE` was set to `servicex_databinder`, the input data has already been pre-processed and will be processed further with `coffea`.
+# What happens here depends on the flag `USE_SERVICEX`. If set to true, the processor is run on the data previously gathered by  ServiceX, then will gather output histograms.
+#
+# When `USE_SERVICEX` is false, the input files need to be processed during this step as well.
 
 # %%
-if PIPELINE == "coffea":
-    if USE_DASK:
-        executor = processor.DaskExecutor(client=utils.get_client(AF))
-    else:
-        executor = processor.FuturesExecutor(workers=NUM_CORES)
-
-    run = processor.Runner(executor=executor, schema=AGCSchema, savemetrics=True, metadata_cache={}, chunksize=CHUNKSIZE)
-
-    filemeta = run.preprocess(fileset, treename="events")  # pre-processing
-
-    t0 = time.monotonic()
-    all_histograms, metrics = run(fileset, "events", processor_instance=TtbarAnalysis(DISABLE_PROCESSING, IO_FILE_PERCENT))  # processing
-    exec_time = time.monotonic() - t0
-    all_histograms = all_histograms["hist"]
-
-elif PIPELINE == "servicex_processor":
-    # in a notebook:
-    t0 = time.monotonic()
-    all_histograms = await utils.produce_all_histograms(fileset, get_query, TtbarAnalysis(DISABLE_PROCESSING, IO_FILE_PERCENT),
-                                                        use_dask=USE_DASK, ignore_cache=SERVICEX_IGNORE_CACHE, schema=AGCSchema)
-    exec_time = time.monotonic() - t0
-
-    # as a script:
-    # async def produce_all_the_histograms():
-    #     return await utils.produce_all_histograms(fileset, get_query, TtbarAnalysis(DISABLE_PROCESSING, IO_FILE_PERCENT),
-    #                                               use_dask=USE_DASK, ignore_cache=SERVICEX_IGNORE_CACHE, schema=AGCSchema)
-    #
-    # all_histograms = asyncio.run(produce_all_the_histograms())
-
-elif PIPELINE == "servicex_databinder":
-    # needs a slightly different schema, not currently implemented
-    raise NotImplementedError("further processing of this method is not currently implemented")
+if USE_DASK:
+    executor = processor.DaskExecutor(client=utils.get_client(AF))
+else:
+    executor = processor.FuturesExecutor(workers=NUM_CORES)
+        
+run = processor.Runner(executor=executor, schema=AGCSchema, savemetrics=True, metadata_cache={}, chunksize=CHUNKSIZE)
+
+if USE_SERVICEX:
+    treename = "servicex"
+    fileset = fileset_queried
+    
+else:
+    treename = "events"
+    
+filemeta = run.preprocess(fileset, treename=treename)  # pre-processing
+
+t0 = time.monotonic()
+all_histograms, metrics = run(fileset, treename, processor_instance=TtbarAnalysis(DISABLE_PROCESSING, IO_FILE_PERCENT))  # processing
+exec_time = time.monotonic() - t0
+
+all_histograms = all_histograms["hist"]
 
 print(f"\nexecution took {exec_time:.2f} seconds")
 
 # %%
 # track metrics for pure coffea setups
-if PIPELINE == "coffea":
+if not USE_SERVICEX:
```

Comment:
We don't need this check anymore, it previously existed since the ServiceX processor did not write out metrics. Now that we're using the same `coffea` approach either way, we always get metrics. Let's add the `USE_SERVICEX` setting to the metrics as well for tracking purposes.

Reply:
Addressed in [d2ab835](https://github.com/iris-hep/analysis-grand-challenge/pull/103/commits/d2ab83532ae72e5ecd07f3f0702a442f26f95842)